pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -125,9 +127,9 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 		}
 
 		clusterImage := cephCluster.Spec.CephVersion.Image
-		cephVersion, ok := getImageVersion(clusterImage)
-		if !ok {
-			logger.Warningf("ceph version not found for image %q used by cluster %q", clusterImage, cephCluster.Name)
+		cephVersion, err := getImageVersion(cephCluster)
```

Comment:
As a question for @egafford or @leseb, will this `reconcile` method definintely get called after the orchestration loop has run its new ceph version check? Like, when a cluster is updated or when the operator is restarted, will this definitely always run and run **after** the ceph version is checked?

Reply:
There's a (5 minute) retry loop in the getImageVersion function to ensure that the new version will be available in the case of a race condition during upgrade.

Reply:
I'm trying to understand how this information fits as a comforting answer to my question. Which is not to say that it isn't the right info. I think I am missing more information. 

1. The operator starts, or an orchestration loop is started
2. getImageVersion is called
...
A. this reconcile loop is called

How can we absolutely certain that (A) will not be called before (2) is complete?

Reply:
We can't, the reconcile loop will start anyway which makes me think that we should return a noop here if we can't get the version, so what about `return reconcile.Result{}, nil` after logging in the error condition?

Reply:
So first, getImageVersion is part of this reconcile loop (it's called on line 129). detectAndValidateCephVersion may be triggered before or after this reconcile loop, but that's why we have the poll in place to make sure that the CephCluster has an up to date image version before we move on. 

Still, I'm okay returning reconcile.Result{}, especially given that I see that crash.go is now depending on a non-nil CephVersion. I'm a little bit concerned here that if there are multiple namespaced CephClusters with pods on the same node, we'd be blocking all clusters for an error in one, but looking around the code it seems that pattern is pretty endemic. We may want to look at continuing the namespace loop and aggregating errors in a separate patch. Hopefully this won't be a common case. For now, doing this thing.